File: 프로그래머스/lv3/43238. 입국심사/입국심사.js
function solution(n, times) {
    let l=1,r=1000000000000000000;
    while(l<r){
        let mid=Math.floor((l+r)/2);
        let now=0;
        times.forEach(a=>now+=Math.floor(mid/a))
        if(now<n){
            l=mid+1;
        }
        else{
            r=mid;
        }
    }
    return l;
}

File: 프로그래머스/lv3/43238. 입국심사/입국심사.cs
using System;

public class Solution {
    public long solution(int n, int[] times) {
        long answer = 0;
        long l=1,r=1000000000000000000;
        while(l<r){
            long mid=(l+r)>>1;
            long now=0;
            foreach(int k in times){
                now+=mid/k;
            }
            if(now<n){
                l=mid+1;
            }
            else{
                r=mid;
            }
        }
        return l;
    }
}